### PR TITLE
Added methods to pull series data to DataCommonsClient

### PR DESCRIPTION
**Pull request by dwnoble**
Example on Observable: https://observablehq.com/d/583a958bacfc3f71

Screenshot 2024-01-12 at 3 35 35 PM

### CODE REVIEW

File: packages/client/src/utils.ts [hunk 1]
```diff
@@ -73,3 +76,49 @@ export function encodeCsvRow(items: any[]): string {
     })
     .join(",");
 }
+
+/**
+ * Given a observation series `num`, computes the ratio of the each
+ * observation value in `num` to the observation value in `denom` with
+ * the closest date value.
+ *
+ * Both `num` and `denom` series are sorted.
```
Comment (juliawu) [inline packages/client/src/utils.ts:85]: Are these sorted by date (as opposed to value)? Please specify in this spec.

File: packages/client/src/utils.ts [hunk 2]
```diff
@@ -73,3 +76,49 @@ export function encodeCsvRow(items: any[]): string {
...
+): Observation[] {
+  if (_.isEmpty(denom)) {
+    return [];
+  }
+  const result: Observation[] = [];
+  let j = 0; // denominator position
+  for (let i = 0; i < num.length; i++) {
+    const numDate = Date.parse(num[i].date);
+    const denomDate = Date.parse(denom[j].date);
+    while (j < denom.length - 1 && numDate > denomDate) {
```
Comment (juliawu) [inline packages/client/src/utils.ts:105]: nit: add a quick comment on what the while loop does, so it's easier for future maintainers to parse this function at a glance. e.g. `// find denom with closest date to num`

File: packages/client/src/data_commons_client_types.ts [hunk 3]
```diff
@@ -32,6 +32,13 @@ export interface BaseGetDataRowsParams {
    * Must be a subset of `variables` param.
    */
   perCapitaVariables?: string[];
+  /**
+   * Delimiter for field header.
+   * Example, if fieldDelimiter = ".", entity value header will be "entity.value"
+   *
+   * Default: "__"
+   */
+  fieldDelimiter?: string;
```
Comment (juliawu) [inline packages/client/src/data_commons_client_types.ts:41]: Perhaps we should add this to the documentation in the readme?

**Review by juliawu — APPROVED** [review]
Awesome! Just a few nits.

**Review by chejennifer — APPROVED** [review]
nice, just some nits!

File: packages/client/src/utils.ts (hunk 1 above)
Comment (dwnoble) [reply packages/client/src/utils.ts:85]: done!

File: packages/client/src/utils.ts (hunk 2 above)
Comment (dwnoble) [reply packages/client/src/utils.ts:105]: Done!

File: packages/client/src/data_commons_client_types.ts (hunk 3 above)
Comment (dwnoble) [reply packages/client/src/data_commons_client_types.ts:41]: Good call-added.

**Pull request merged**